d2bs/kolbot/libs/modules/Town.js

```diff
@@ -352,6 +352,31 @@
 		return true;
 	};
 
+	// Purchases and drinks potions in town.
+	Town.stackPotions = function(type, amount) {
+		if (!amount) amount = 4; // TODO: Calculate monster effort and correlate it to the number of potions to stack.
+		if (me.gold < amount * 25) {
+			print('stackPotions: Not enough gold for ' + amount + ' potions');
+			return false;
+		}
+		const npc = Town.initNPC("Shop", "buyPotions");
+		for (var i = 0; i < amount; i++) {
+			const npcPotion = Town.getPotion(npc, type);
+			if (npcPotion) {
+				npcPotion.buy(false);
+			}
+		}
+		me.getItems(npcPotion.classid)
+			.filter(potion => (potion.classid == npcPotion.classid && (potion.mode === sdk.itemmode.inBelt || potion.location === sdk.storage.Inventory)))
+			.every((potion, index) => {
```

Comment:
Use `every` here so that we can exit early (`return false`). I was surprised to learn we cannot exit early (`break`) with `forEach`. So it's either iterating with a `for` or `every`. YMMV.